packages/web-client/app/controllers/card-pay/balances.ts

```diff
@@ -2,16 +2,31 @@ import Controller from '@ember/controller';
 import { tracked } from '@glimmer/tracking';
 import { inject as service } from '@ember/service';
 import Layer2Network from '@cardstack/web-client/services/layer2-network';
+import { action } from '@ember/object';
+import RouterService from '@ember/routing/router-service';
 
 class CardPayBalancesController extends Controller {
   @service declare layer2Network: Layer2Network;
+  @service declare router: RouterService;
 
-  queryParams = ['flow'];
-  @tracked flow: 'deposit' | null = null;
+  queryParams = ['flow', { workflowPersistenceId: 'flow-id' }];
+  @tracked flow: string | null = null;
+  @tracked workflowPersistenceId: string | null = null;
 
   get prepaidCards() {
     return this.layer2Network.safes.value?.filterBy('type', 'prepaid-card');
   }
+
+  @action setFlow(flow: string) {
```

Comment:
Maybe we should rename this method since I have changed the implementation?

Reply:
Good catch. I think `transitionToBalances` would be more suitable.